sdk/src/loom/Platform/GameController.ls

```diff
@@ -0,0 +1,160 @@
+        public static var BUTTON_GUIDE:uint         =  5;
+        public static var BUTTON_START:uint         =  6;
+        public static var BUTTON_LEFTSTICK:uint     =  7;
+        public static var BUTTON_RIGHTSTICK:uint    =  8;
+        public static var BUTTON_LEFTSHOULDER:uint  =  9;
+        public static var BUTTON_RIGHTSHOULDER:uint = 10;
+        public static var BUTTON_DPAD_UP:uint       = 11;
+        public static var BUTTON_DPAD_DOWN:uint     = 12;
+        public static var BUTTON_DPAD_LEFT:uint     = 13;
+        public static var BUTTON_DPAD_RIGHT:uint    = 14;
+        /**
+         * BUTTON_COUNT is not a valid button by itself
+         * It is used to help iterating through buttons
+         */
+        public static var BUTTON_MAX:uint = 15;
+        
+        /**
+         * Helper variables for identifying axes
+         */
+        public static var AXIS_LEFTX:uint        = 0;
+        public static var AXIS_LEFTY:uint        = 1;
+        public static var AXIS_RIGHTX:uint       = 2;
+        public static var AXIS_RIGHTY:uint       = 3;
+        public static var AXIS_TRIGGERLEFT:uint  = 4;
+        public static var AXIS_TRIGGERRIGHT:uint = 5;
+        /**
+         * AXIS_COUNT is not a valid axis by itself
+         * It is used to help iterating through axes
+         */
+        public static var AXIS_MAX:uint = 6;
+        
+        /** Specifies the number of connected game controllers. */
+        public static native var numControllers:int;
+        
+        /** Name of a connected controller. */
+        public native var name:String;
+        
+        /** Delegate for handling button events. */
+        public native var onButtonEvent:ButtonDelegate;
+        /** Delegate for handling axis events. */
+        public native var onAxisEvent:AxisDelegate;
+        
+        /**
+         * Returns a GameController object.
+         * 
+         * @param index The index of the controller
+         * @return Returns a GameController if it is connected
+         */
+        public static native function getGameController(index:int = -1):GameController;
+        
+        /**
+         * Checks if game controller is haptic.
+         * 
+         * @param device The id of game controller.
+         * @return True if game controller is haptic, false if it is not haptic.
+         */
+        public native function isHaptic():Boolean;
+        
+        /**
+         * Causes a game controller to start vibrating.
+         * 
+         * @param device The id of game controller.
+         * @param intensity Intensity of vibration ranging from 0 to 1.
+         * @param ms Duration of vibration in miliseconds. Accepts GameController.RUMBLE_INFINITY for infinite duration.
+         */
+        public native function startRumble(intensity:Number, ms:uint);
+        
+        /**
+         * Causes a game controller to stop vibrating.
+         * 
+         * @param device The id of game controller.
```

Comment:
Looks like you list device as a param in many cases here but it's now instance based so not passed?